ksql-engine/src/test/java/io/confluent/ksql/planner/plan/JoinNodeTest.java

```diff
@@ -238,4 +302,619 @@ public void shouldHaveAllFieldsFromJoinedInputs() {
+                                           rightKeyFieldName,
+                                           leftAlias,
+                                           rightAlias,
+                                           null,
+                                           DataSource.DataSourceType.KSTREAM,
+                                           DataSource.DataSourceType.KSTREAM);
+
+    try {
+      joinNode.buildStream(mockStreamsBuilder,
+                           mockKsqlConfig,
+                           mockKafkaTopicClient,
+                           mockFunctionRegistry,
+                           properties,
+                           mockSchemaRegistryClient);
+      fail("Should have raised an exception since no join window was specified");
+    } catch (KsqlException e) {
+      // good;
+    }
+
+    verify(left, right, leftSchemaKStream, rightSchemaKStream);
+
+    assertEquals(leftKeyFieldName, joinNode.getLeftKeyFieldName());
+    assertEquals(rightKeyFieldName, joinNode.getRightKeyFieldName());
+    assertEquals(leftAlias, joinNode.getLeftAlias());
+    assertEquals(rightAlias, joinNode.getRightAlias());
+    assertEquals(JoinNode.JoinType.INNER, joinNode.getJoinType());
+  }
+
+
+  @SuppressWarnings("unchecked")
+  @Test
+  public void shouldNotPerformJoinIfInputPartitionsMisMatch() {
+    expect(left.getSchema()).andReturn(leftSchema);
+    expect(left.getPartitions(mockKafkaTopicClient)).andReturn(3);
+
+    expect(right.getSchema()).andReturn(rightSchema);
+    expect(right.getPartitions(mockKafkaTopicClient)).andReturn(2);
+
+    expectSourceName(left);
+    expectSourceName(right);
+    final SpanExpression spanExpression = new SpanExpression(10, TimeUnit.SECONDS);
+
+    replay(left, right, leftSchemaKStream, rightSchemaKStream);
+
+    final JoinNode joinNode = new JoinNode(new PlanNodeId("join"),
+                                           JoinNode.JoinType.OUTER,
+                                           left,
+                                           right,
+                                           leftKeyFieldName,
+                                           rightKeyFieldName,
+                                           leftAlias,
+                                           rightAlias,
+                                           spanExpression,
+                                           DataSource.DataSourceType.KSTREAM,
+                                           DataSource.DataSourceType.KSTREAM);
+
+    try {
+      joinNode.buildStream(mockStreamsBuilder,
+                           mockKsqlConfig,
+                           mockKafkaTopicClient,
+                           mockFunctionRegistry,
+                           properties,
+                           mockSchemaRegistryClient);
+      fail("should have raised an exception since the number of partitions on the input sources "
+           + "don't match");
+    } catch (KsqlException e) {
+      // good!
```

Comment:
maybe do some validation of the message to make sure we're throwing for the right reason.